src/samVisualize/graphGeneration/cfnTemplateGraphGenerator.ts

```diff
@@ -0,0 +1,135 @@
+            case LinkTypes.GetAtt:
+                graph.createLink(parentNodeName, value[0], LinkTypes.GetAtt)
+                break
+
+            //  Adding a single link, from a "Ref" key
+            //  https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ref.html
+            case LinkTypes.Ref:
+                graph.createLink(parentNodeName, value, LinkTypes.Ref)
+                break
+
+            //  Extracting an link out of a substitution
+            //  A sub link can point to an array, or a single string.
+            //  We can immediately extract the substituion if it points to a string, otherwise we must continue traversing
+            //  https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html
+            case LinkTypes.Sub:
+                if (_.isString(value)) {
+                    const substitutions = extractSubstitution(value)
+                    for (const destNodeName of substitutions) {
+                        graph.createLink(parentNodeName, destNodeName, LinkTypes.Sub)
+                    }
+                } else {
+                    // Key must point to an array
+                    for (const element of value) {
+                        traverse(graph, element, parentNodeName)
+                    }
+                }
+                break
+            //  If a key is not a possible link, we continue to traverse for nested links
+            default:
+                // If the value is an array, we traverse each traversable element
+                if (Array.isArray(value)) {
+                    for (const element of value) {
+                        traverse(graph, element, parentNodeName)
+                    }
+                } else {
+                    traverse(graph, value, parentNodeName)
+                }
+        }
+    }
+}
+/**
+ * @param inputYaml A string representing a YAML template
+ * @returns A GraphObject representing the graph
+ */
+function generateGraphFromYaml(inputYaml: string): GraphObject {
+    const templateData: Record<string, any> = yamlStringToObject(inputYaml)
+    // We only want Resources as nodes in our graph
+    const resources = templateData['Resources']
+    const graph = new Graph()
+
+    // Loop over each resource and initialize it in the map.
+    // These are the only nodes we wanna be working with, links to non-resources will be ignored.
+    for (const resourceName of Object.keys(resources)) {
+        graph.initNode(resourceName, resources[resourceName]['Type'])
+    }
+    // Now we can traverse each resource, after having defined each possible node in the map
+    for (const resourceName of Object.keys(resources)) {
+        graph.initNode(resourceName, resources[resourceName]['Type'])
+        traverse(graph, resources[resourceName], resourceName)
+    }
+    // Return
+    return graph.getObjectRepresentation()
+}
+
+export { generateGraphFromYaml }
```

Comment:
We typically don't have an extra export statement like this. Instead, add `export` to the function declaration:

```
export function generateGraphFromYaml(inputYaml: string): GraphObject {
```